File: C++/Basic/input.cpp
#include <iostream>
using namespace std;

int main() {
    const int year = 2023;
    int birth_year;

    cout << "Pass your birth year: ";
    cin >> birth_year;

    if (cin.fail() || birth_year < 1900 || birth_year > year) {
        cout << "Wrong Input";
        return 0;
    }

    cout << "You will be " << year - birth_year << " by the end of this year :]";
    return 0;
}

File: Golang/Basic/input.go
package main

import (
	"fmt"
	"bufio"
	"os"
	"strconv"
)

func input() {
	// Get user input
	scanner := bufio.NewScanner(os.Stdin)
	fmt.Printf("Type something: ")
	scanner.Scan()

	input := scanner.Text()
	fmt.Printf("You typed: %q\n\n", input)

	
	// Check user age
	fmt.Printf("Type your birth year: ")
	scanner.Scan()

	age, _ := strconv.ParseInt(scanner.Text(), 10, 64)
	fmt.Printf("You will be %d by the end of 2023", 2023 - age)
}